File: Day_3/java/Day3Loops.java
import java.util.Scanner;

public class Day3Loops {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.println("Day 3 Loop Problems");
        System.out.println("1. Print numbers from 1 to N");
        System.out.println("2. Print even numbers from 1 to N");
        System.out.println("3. Sum of first N natural numbers");
        System.out.println("4. Factorial of a number");
        System.out.println("5. Count digits in a number");
        System.out.print("Choose a problem (1-5): ");

        int choice = sc.nextInt();

        if (choice == 1) {
            System.out.print("Enter n: ");
            int n = sc.nextInt();
            for (int i = 1; i <= n; i++) {
                System.out.println(i);
            }
        } else if (choice == 2) {
            System.out.print("Enter n: ");
            int n = sc.nextInt();
            for (int i = 1; i <= n; i++) {
                if (i % 2 == 0) {
                    System.out.println(i);
                }
            }
        } else if (choice == 3) {
            System.out.print("Enter n: ");
            int n = sc.nextInt();
            int total = 0;
            for (int i = 1; i <= n; i++) {
                total += i;
            }
            System.out.println("Sum: " + total);
        } else if (choice == 4) {
            System.out.print("Enter n: ");
            int n = sc.nextInt();
            int fact = 1;
            for (int i = 1; i <= n; i++) {
                fact *= i;
            }
            System.out.println("Factorial: " + fact);
        } else if (choice == 5) {
            System.out.print("Enter number: ");
            int n = sc.nextInt();
            n = Math.abs(n);
            int count = 0;
            if (n == 0) {
                count = 1;
            } else {
                while (n > 0) {
                    count++;
                    n /= 10;
                }
            }
            System.out.println("Digits: " + count);
        } else {
            System.out.println("Invalid choice");
        }

        sc.close();
    }
}

File: Day_3/python/day3_loops.py
"""Day 3 loop-based practice programs."""


def print_1_to_n():
    n = int(input("Enter n: "))
    for i in range(1, n + 1):
        print(i)


def print_even_1_to_n():
    n = int(input("Enter n: "))
    for i in range(1, n + 1):
        if i % 2 == 0:
            print(i)


def sum_first_n():
    n = int(input("Enter n: "))
    total = 0
    for i in range(1, n + 1):
        total += i
    print("Sum:", total)


def factorial_n():
    n = int(input("Enter n: "))
    fact = 1
    for i in range(1, n + 1):
        fact *= i
    print("Factorial:", fact)


def count_digits():
    n = int(input("Enter number: "))
    n = abs(n)
    count = 0
    if n == 0:
        count = 1
    else:
        while n > 0:
            count += 1
            n //= 10
    print("Digits:", count)


print("Day 3 Loop Problems")
print("1. Print numbers from 1 to N")
print("2. Print even numbers from 1 to N")
print("3. Sum of first N natural numbers")
print("4. Factorial of a number")
print("5. Count digits in a number")

choice = int(input("Choose a problem (1-5): "))

if choice == 1:
    print_1_to_n()
elif choice == 2:
    print_even_1_to_n()
elif choice == 3:
    sum_first_n()
elif choice == 4:
    factorial_n()
elif choice == 5:
    count_digits()
else:
    print("Invalid choice")

